### SQL005

**Using 'SaveChanges' method in a loop can affect performance**

Noncompliant Code Example:  
```csharp
var dbContext = new DbContext("test");
for (int i = 0; i < 100; i++)
{
    dbContext.Entities.Add(new Entity(i));
    s.SaveChanges();
}
```

Compliant Solution:  
```csharp
var dbContext = new DbContext("test");
for (int i = 0; i < 100; i++)
{
    dbContext.Entities.Add(new Entity(i));
}

s.SaveChanges();
```
