### SQL003
**Using 'Query' method is not optimal here**

Noncompliant Code Example:  
```csharp
var dog = connection.Query<Dog>("select * from dogs").Single();
```

Compliant Solution:  
```csharp
var dog = connection.QuerySingle<Dog>("select * from dogs");
```

https://github.com/StackExchange/Dapper#performance
